backend/open_webui/models/prompts.py

```
@@ -155,5 +313,127 @@ def delete_prompt_by_command(self, command: str) -> bool:
+            # Get the filtered prompts (much smaller set now)
+            prompts = query.all()
+
+            # For the small result set, check shared prompts with has_access
+            accessible_count = 0
+            for prompt in prompts:
+                # Public prompts (access_control is stored as string 'null') are accessible to everyone
+                if prompt.access_control == "null" or prompt.access_control is None:
+                    accessible_count += 1
+                # User's own prompts are always accessible
+                elif prompt.user_id == user_id:
+                    accessible_count += 1
```

Ruff (0.12.2)

Combine `if` branches using logical `or` operator

Combine `if` branches

(SIM114)